### ISSUE TITLE
`cx test` aws doesn't run test commands properly

### ISSUE THREAD

**Issue by nzoschke** [convox/praxis#225]
Running `cx test` against aws panics. We tracked this down to the fact that a build and release is created, but the release is not promoted and therefore `cx run` doesn't work. It looks related to not having a task definition.

Locally we don't need to promote the release to run a one-off.

However I think we should consider promoting the release in both cases.

I have an app that uses postgres both for production and testing. It seems like a promote and a run that has DATABASE_URL is a universal solution for this pattern. However this pattern is blocked on another issue https://github.com/convox/praxis/issues/165

```
build   | Successfully built dd2f16492d10
build   | running: docker tag 9836064b94124bad54f83c70026dd85fcb8b5a13 staging-test-1496978927/proxy:BUMBJGRRZN
build   | running: docker tag 9836064b94124bad54f83c70026dd85fcb8b5a13 staging-test-1496978927/web:BUMBJGRRZN
build   | running: docker tag staging-test-1496978927/proxy:BUMBJGRRZN 782231114432.dkr.ecr.us-east-1.amazonaws.com/stagi-repos-nxp774qw2q30:proxy.BUMBJGRRZN
build   | pushing: 782231114432.dkr.ecr.us-east-1.amazonaws.com/stagi-repos-nxp774qw2q30:proxy.BUMBJGRRZN
web     | running: bin/test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x408a60f]

goroutine 2760 [running]:
io.copyBuffer(0x5440e00, 0xc42016cbd0, 0x0, 0x0, 0xc42060e000, 0x8000, 0x8000, 0x0, 0xc420cbedb0, 0x2)
    /usr/local/go/src/io/io.go:390 +0xdf
io.Copy(0x5440e00, 0xc42016cbd0, 0x0, 0x0, 0x69, 0xc42008dbc0, 0x405c8e2)
    /usr/local/go/src/io/io.go:360 +0x68
created by github.com/convox/praxis/sdk/rack.(*Client).Stream
    /ext-go/1/src/github.com/convox/praxis/sdk/rack/client.go:130 +0x60c
```
